### ISSUE TITLE
Does not work on macOS x86_64, or Linux x86_64 due to bug in FFI

### ISSUE THREAD

**Issue by ape-dev-cs** [omurovec/foundry-zksync#7]
When using this on non Apple Silicon macOS or on x64 Linux, foundry-zksync-era will download the macOS ARM binary & therefore not function as intended.

This is due to a bug in the way FFI has been implemented to detect system info. It just gets back `$(uname -m)` verbatim from the ffi call, without evaluating the intended command, hence the ternary statements default to `arch` and `os` to macOS / arm64 respectively, which leads to the wrong binary being downloaded & executed.

**Comment by tonisives** [comment]
same
```
    │   ├─ [0] VM::ffi([curl, -L, https://github.com/matter-labs/zksolc-bin/raw/main/macosx-arm64/zksolc-macosx-arm64-v1.3.4, --output, /Users/tonis/workspace/eth/playground/forge-zkzync/lib/zksolc-macosx-arm64-v1.3.4, --silent])
    │   │   └─ ← 0x
    │   ├─ [0] VM::ffi([chmod, +x, /Users/tonis/workspace/eth/playground/forge-zkzync/lib/zksolc-macosx-arm64-v1.3.4])
    │   │   └─ ← 0x
    │   └─ ← 4740 bytes of code
    ├─ [4380] Deployer::deployFromL1(src/Counter.sol, 0x, 0x0000000000000000000000000000000000000000000000000000000000000539, true)
    │   ├─ [0] VM::ffi([/Users/tonis/workspace/eth/playground/forge-zkzync/lib/zksolc-macosx-arm64-v1.3.4, --bin, src/Counter.sol])
    │   │   └─ ← "Failed to execute command: Bad CPU type in executable (os error 86)"
```

**Comment by omurovec** [comment]
This PR should fix it: #9 

I'm not able to test very easily so please let me know if this fix works for you and I'll merge 
@ape-dev-cs @tonisives

**Comment by th3m477** [comment]
Tested this on my intel mac, works @omurovec  👍
